physics/horizontal_projectile_motion.py

```diff
@@ -0,0 +1,122 @@
+"""
+Horizontal Projectile Motion problem in physics.
+This algorithm solves a specific problem in which
+the motion starts from the ground as can be seen below:
+      (v = 0)
+        **
+       *  *
+      *    *
+     *      *
+    *        *
+   *          *
+GROUND      GROUND
+
+For more info: https://en.wikipedia.org/wiki/Projectile_motion
+"""
+
+# Importing packages
+from math import radians as angle_to_radians
+from math import sin
+
+# Acceleration Constant on hearth (unit m/s^2)
+g = 9.80665
+
+
+def horizontal_distance(init_velocity: float, angle: float) -> float:
+    """
+    Returns the horizontal distance that the object cover
+
+    Formula:
+            v_0^2 * sin(2 * alpha)
+            ---------------------
+                   g
+
+    v_0 - initial velocity
+    alpha - angle
+
+    >>> horizontal_distance(30, 45)
```

Comment:
Tests with floats, zeros, negative numbers, etc.?

Reply:
Algorithms should throw exceptions when they get bad data.  Don't push that effort on your caller but do it in the function so that garbage input does not pass.